src/components/Legend/Legend.tsx

```diff
@@ -0,0 +1,237 @@
+import _ from 'lodash';
+import React from 'react';
+import PropTypes from 'react-peek/prop-types';
+import { lucidClassNames } from '../../util/style-helpers';
+import {
+	FC,
+	findTypes,
+	omitProps,
+	StandardProps,
+} from '../../util/component-types';
+import Point from '../Point/Point';
+import Line from '../Line/Line';
+import { any } from 'prop-types';
+
+const cx = lucidClassNames.bind('&-Legend');
+
+const POINT_SIZE = 12;
+const LINE_WIDTH = 22;
+
+const { number, string, oneOf, node, bool, func } = PropTypes;
+
+// Default props for the Legend component
+const defaultProps = {
+	orient: 'vertical',
+	hasBorders: true,
+	isReversed: false,
+};
+
+export interface ILegendProps
+	extends StandardProps,
+		React.DetailedHTMLProps<
+			React.HTMLAttributes<HTMLDivElement>,
+			HTMLDivElement
+		> {
+	/** Custom Item element (alias for `Legend.Item`) */
+	Item?: React.ReactNode;
+
+	/** Determines if the legend is vertical or horizontal */
+	orient?: string;
```

Comment:
should we make this more explicit like `'vertical' | 'horizontal'` since we don't want to accept any random string?

Reply:
I like that idea. Consider it done.